### ISSUE TITLE
[ Feature Request ] In Code and General Documentation

### ISSUE THREAD

**Issue by PeterHindes** [TeamNewPipe/NewPipe#2725]
There doesn't seem to be an appropriate amount of comments. This is pretty concerning as it means that new contributors (ie. me) may be somewhat lost. I know that this wont improve the functionality of the code directly, but it will make the code more approachable to new contributors. I plan to write some more in code and out of code docs as i become familiar with the project, but I would appreciate some help with this fairly large task. So if you have some free time feel free to work on some docs. 😊

**Comment by PeterHindes** [comment]
I have found the general docs on [the main site](https://teamnewpipe.github.io/documentation/). But i still think the code needs some better docs. IDK, maybe I'm just being picky.

**Comment by opusforlife2** [comment]
Agreed. I found it pretty confusing as well to figure out which function was doing what, and where values were coming from.
